timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -1219,7 +1234,8 @@ void declare_barostat(py::module &m) {
             py::arg("frequency"),
             py::arg("bps"),
             py::arg("seed"),
-            py::arg("adaptive_scaling_enabled") = true)
```

Comment:
Is there a need to remove this default? It means any old code that constructs a barostat needs to be changed, even though the default behavior won't have changed

Reply:
My idea was to have a single source of truth for defaults, to avoid specifying them here and also in `timemachine.lib.MonteCarloBarostat`, which seemed like it might be confusing and get out of sync. The reason to prefer setting them in the Python API was to allow making arguments optional, so the user can pass `None` to use the default value.

This should only require changing code that calls `custom_ops.MonteCarloBarostat` directly; I think in most cases we should be using the Python API wrapper, `timemachine.lib.MonteCarloBarostat`? (usage of the latter should not need changes)

Reply:
That seems like a reasonable rationale. Would prefer in the future that these sorts of changes be made in separate PRs where that intention is communicated rather than at the tail of PR review. Was surprising to see you change this after approvals as it does change how the code base can be used.

Reply:
Yeah, fair, sorry about that. This wasn't a change out of the blue, but was intended to address https://github.com/proteneer/timemachine/pull/1151#discussion_r1336112633.

Reply:
I understand the intent of the change, however the scope of the change is beyond what the two comments implied to me.

No action needs to be taken in this case, and for the future would be good to avoid such changes at this stage of a PR or to more clearly point out this sort of API change.

Reply:
OK, cool. I'm also happy to add the defaults back to `wrap_kernels.cpp`, and isolate the API change in a separate PR if that sounds preferable.

Reply:
Removing defaults is probably fine for this PR - I'd actually much rather this be made explicit in this PR (since we're clearly changing default behavior).

Reply:
Added an item to the PR description